src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+   * 
+   * @param src Grid function on (refined) redecomp mesh
+   * @param dst Zero-valued grid function on (coarse) submesh
+   */
+  void RedecompToSubmesh(
+    const mfem::GridFunction& src,
+    mfem::ParGridFunction& dst
+  ) const;
+
+  /**
+   * @brief Get the (coarse) Submesh associated with the SubmeshRedecompTransfer object
+   * 
+   * @return const mfem::ParSubMesh& 
+   */
+  const mfem::ParSubMesh& GetSubmesh() const 
+  { 
+    return static_cast<const mfem::ParSubMesh&>(*submesh_fes_.GetParMesh());
+  }
+
+  /**
+   * @brief Set a new (refined) redecomp mesh associated with the existing (refined) submesh
+   * 
+   * @param redecomp RCB redecomposed (refined) submesh
+   */
+  void UpdateRedecomp(redecomp::RedecompMesh& redecomp);
+
+private:
+  /**
+   * @brief Create a finite element space on the (refined) redecomp mesh
+   * 
+   * @param redecomp RCB redecomposed (refined) submesh
+   * @param submesh_fes Original finite element space on the (refined) submesh
+   * @return std::unique_ptr<mfem::FiniteElementSpace> 
+   */
+  static std::unique_ptr<mfem::FiniteElementSpace> CreateRedecompFESpace(
+    redecomp::RedecompMesh& redecomp,
+    mfem::ParFiniteElementSpace& submesh_fes
+  );
+
+  /**
+   * @brief Finite element space on the (coarse) submesh 
+   */
+  mfem::ParFiniteElementSpace& submesh_fes_;
+
+  /**
+   * @brief Finite element space on the (refined) redecomp mesh
+   */
+  mutable std::unique_ptr<mfem::FiniteElementSpace> redecomp_fes_;
+
+  /**
+   * @brief Transfer object between low-order and higher-order grid functions
+   */
+  SubmeshLORTransfer* lor_xfer_;
+
+  /**
+   * @brief Transfer object between (refined) submesh and (refined) redecomp mesh
+   */
+  const redecomp::RedecompTransfer redecomp_xfer_;
+};
+
+/**
+ * @brief Facilitates transferring variables from the original mesh to redecomp
+ * levels
+ *
+ * This class simplifies transferring field variables from/to a grid function on
+ * an mfem::ParMesh to/from a grid function on a redecomp::RedecompMesh.  If
+ * transferring to a low-order grid function on a refined mesh is also required,
+ * this class will perform that transfer as well.
+ */
+class ParentRedecompTransfer
+{
+public:
+  /**
+   * @brief Construct a new ParentRedecompTransfer object
+   *
+   * @param parent_fes Finite element space on the original (coarse) mesh
+   * @param submesh_gridfn Grid function on the (coarse) submesh used to
+   * temporarily store variables being transferred
+   * @param lor_xfer (Optional) Low-order grid function, refined mesh transfer
+   * object
+   * @param redecomp RCB redecomposed (refined) submesh
+   */
+  ParentRedecompTransfer(
+    const mfem::ParFiniteElementSpace& parent_fes,
+    mfem::ParGridFunction& submesh_gridfn,
+    SubmeshLORTransfer* lor_xfer,
+    redecomp::RedecompMesh& redecomp
+  );
+
+  /**
+   * @brief Transfer grid function on original mesh to grid function on redecomp
```

Comment:
Avoid words like "original". Also avoid coarse below and refined. If parent mesh is k=1, then redecomp mesh is also k=1 with equal coarseness/fineness, so these words don't mean much if I'm reading this correctly. If parent mesh is k>1, then we need LOR, and indeed the redecomp mesh will be some finer redecomposed mesh.  I think you keep:

@param src Par Grid function on parent mesh
@return mfem::GridFunction on redecomp mesh

@note if parent mesh is HO, LOR is used resulting in a refined redecomp mesh

Or something like that. We have the two cases k=1, and k>1 we have to speak to

Reply:
Updated